src/safeds_stubgen/stubs_generator/_generate_stubs.py

```diff
@@ -161,12 +161,27 @@ def _create_class_string(self, class_: Class, class_indentation: str = "") -> st
         # Superclasses
         superclasses = class_.superclasses
         superclass_info = ""
+        superclass_methods_text = ""
         if superclasses and not class_.is_abstract:
             superclass_names = []
             for superclass in superclasses:
-                superclass_names.append(self._split_import_id(superclass)[1])
+                superclass_name = self._split_import_id(superclass)[1]
+                superclass_names.append(superclass_name)
                 self._add_to_imports(superclass)
 
+                # if the superclass was not added to the module_imports through the _add_to_imports method, it means
+                # that the superclass is a class from the same module. We have to check if it's a private class and
+                # add the public classes, if it's an intern class.
+                if superclass not in self.module_imports:
+                    for module_class in self.module.classes:
+                        if module_class.name == superclass_name and superclass_name.startswith("_"):
```

Comment:
This is unnecessarily inefficient. Finding a declaration by name should be fast, e.g. by storing the declarations in a dictionary that maps names to declarations (or lists of declarations depending on how mypy handles non-unique names).

Reply:
You can also pull the `superclass_name.startswith` check out of the loop into the condition of the containing if-statement.